core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -404,16 +404,39 @@ private Pattern validateRegexPattern(String regex, String methodName) {
       }
     }
 
-    /** Helper for multiple capturing group regex check in REGEXP_EXTRACT fns. */
+    /** Helper for multiple capturing group regex check in REGEXP_* fns. */
     private void checkMultipleCapturingGroupsInRegex(Matcher matcher, String methodName) {
       if (matcher.groupCount() > 1) {
         throw RESOURCE.multipleCapturingGroupsForRegexpExtract(
             Integer.toString(matcher.groupCount()), methodName).ex();
       }
     }
 
+    /** Helper for checking values of position and occurrence arguments in REGEXP_* fns.
+     *  Regex Fns not using occurrencePosition param pass a default value as 0.
+     *  Throws an exception or returns true in case of failed value checks. */
+    private boolean checkPosOccurrenceParamValues(int position,
+        int occurrence, int occurrencePosition, String value, String methodName) {
+      if (position <= 0) {
+        throw RESOURCE.invalidIntegerInputForRegexpFunctions(Integer.toString(position),
+            "position", methodName).ex();
+      }
+      if (occurrence <= 0) {
+        throw RESOURCE.invalidIntegerInputForRegexpFunctions(Integer.toString(occurrence),
+            "occurrence", methodName).ex();
+      }
+      if (occurrencePosition < 0 || occurrencePosition > 1) {
+        throw RESOURCE.invalidIntegerInputForRegexpFunctions(Integer.toString(occurrencePosition),
+            "occurrence_position", methodName).ex();
+      }
+      if (position > value.length()) {
+        return true;
+      }
+      return false;
+    }
+
     /** SQL {@code REGEXP_CONTAINS(value, regexp)} function.
-     * Throws a runtime exception for invalid regular expressions.*/
+     *  Throws a runtime exception for invalid regular expressions. */
```

Comment:
Did you mean to add this?

Reply:
yes just to align this line with the ones above